arc-spring-ai-client-adapter-spring-boot-starter/src/main/kotlin/ClientAdapter.kt

```diff
@@ -0,0 +1,66 @@
+// SPDX-FileCopyrightText: 2024 Deutsche Telekom AG
+//
+// SPDX-License-Identifier: Apache-2.0
+
+import io.github.lmos.arc.agents.ArcException
+import io.github.lmos.arc.agents.conversation.AssistantMessage
+import io.github.lmos.arc.agents.conversation.ConversationMessage
+import io.github.lmos.arc.agents.conversation.SystemMessage
+import io.github.lmos.arc.agents.conversation.UserMessage
+import io.github.lmos.arc.agents.functions.LLMFunction
+import io.github.lmos.arc.agents.llm.ChatCompleter
+import io.github.lmos.arc.agents.llm.ChatCompletionSettings
+import io.github.lmos.arc.core.failWith
+import io.github.lmos.arc.core.result
+import org.springframework.ai.chat.ChatClient
+import org.springframework.ai.chat.messages.ChatMessage
+import org.springframework.ai.chat.messages.Message
+import org.springframework.ai.chat.messages.MessageType
+import org.springframework.ai.chat.prompt.ChatOptions
+import org.springframework.ai.chat.prompt.Prompt
+
+private fun messageConversion(msg: ConversationMessage): Message {
+    val role = when (msg) {
+        is UserMessage -> MessageType.USER
+        is AssistantMessage -> MessageType.ASSISTANT
+        is SystemMessage -> MessageType.SYSTEM
+    }
+    return ChatMessage(role, msg.content)
+}
+
+private class SpringAiChatOptionsAdapter(settings: ChatCompletionSettings) : ChatOptions {
+    val settings = settings
```

Comment:
We can remove `val settings = settings` and add private val to the variable about it.